microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java

```diff
@@ -160,6 +160,11 @@ public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completio
 				cancelChecker.checkCanceled();
 
 				CompletionList list = completionResult.getCompletionList();
+				if (list == null) {
+					list = new CompletionList();
+				}
+				CompletionList finalizedList = list;

```

Comment:
this seems unnecessary

Reply:
Yeah, the functional programming makes the logic messy, I'll fix it